text/0000-format-args-implicit-identifiers.md

```diff
@@ -0,0 +1,263 @@
+
+For similar reasons this RFC would argue that introducing a new alternative macro to `format_args!` in the standard library would not be a good outcome compared to adding to the existing macro.
+
+An alternative syntax for implicit named arguments is welcomed by this RFC if it can be argued why it is preferable to the RFC's proposed form. The RFC author argues the chosen syntax is the most suitable, because it matches the existing syntax for named arguments.
+
+
+# Prior art
+[prior-art]: #prior-art
+
+A number of languages support string-interpolation functionality similar to what Rust's formatting macros offer. The RFC author's influence comes primarily from Python 3's "f-strings" and Javscript's backticks.
+
+For a comparison of the three languages, the following code would be the equivalent way to produce a new string combining a `greeting` and a `person`:
+
+    // Rust
+    format!("{} {}", greeting, person)                                // positional form,
+    format!("{greeting} {person}", greeting=greeting, person=person)  // or named form
+
+    # Python 3
+    f"{greeting} {person}"
+
+    // Javascript
+    `${greeting} ${person}`
+
+It is the RFC author's experience that Python and Javascript's functionality read easily from left-to-right and it is clear where each variable is being substituted into the format string.
+
+In the Rust forms illustrated above, the positional form suffers the drawback of not reading strictly from left to right; the reader of the code must refer back-and-forth between the format string and the argument list to determine where each variable will be subsituted. The named form avoids this drawback at the cost of much longer code.
+
+Implementing implicit named arguments in the fashion suggested in this RFC would eliminate the drawbacks of each of the Rust forms and permit new syntax much closer to the other languages:
+
+    // Rust - implicit named arguments
+    format!("{greeting} {person}")
+
+It should be noted, however, that both Python 3's f-strings and Javascript's backticks accept a wide variety of expressions beyond the simple identifier case that this RFC is focussed on. The RFC author argues that supporting expressions inside format strings is unneccessary in Rust, however does not rule out that this is possible as a future extension. Please see the discusison in the [future possibilities](#future-possibilities) section at the end of this RFC.
+
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+## Should implicit named arguments be accepted for formatting parameters?
```

Comment:
How does your PR https://github.com/rust-lang/rust/pull/65338, as currently implemented, answer this question?

Reply:
As implemented that PR would accept implicit named arguments for formatting parameters.

However, I wouldn't necessarily take that as a reason that we should do so; that PR was the smallest code change I could make to play with & propose the feature, with the assumption that refinements would be made from the resulting discussion!

Reply:
If accepting "implicit named arguments" for formatting parameters results in a more consistent spec & overall UX while at the same time requiring less code (the main reason I asked about your PR), then imo then there's little reason to disallow it.